bigcache.go

```diff
@@ -46,20 +47,26 @@ func newBigCache(config Config, clock clock) (*BigCache, error) {
 		config.Hasher = newDefaultHasher()
 	}
 
+	maxShardSize := 0
+	if config.HardMaxCacheSize > 0 {
+		maxShardSize = convertMBToBytes(config.HardMaxCacheSize) / config.Shards
```

Comment:
So it is possible to overwrite data even if total size of cache will be smaller than limit.